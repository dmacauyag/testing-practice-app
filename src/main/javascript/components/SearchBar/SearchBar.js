import React, { Component } from 'react'

class SearchBar extends Component {
  constructor(props) {
    super(props);

    this.handleInputChange = this.handleInputChange.bind(this)
    this.handleKeyPress = this.handleKeyPress.bind(this)
    this.state = {
      searchTerm: ''
    }
  }

  render() {
    return (
      <div className="search-bar">
        <input
          value = {this.state.searchTerm}
          onChange = {event => this.handleInputChange(event)}
          onKeyPress = {event => this.handleKeyPress(event)}
        />
      </div>
    )
  }

  handleInputChange(event) {
    this.setState({
      searchTerm: event.target.value
    })
  }

  handleKeyPress(event) {
    if (event.key === 'Enter') {
      this.props.onSubmitSearch(this.state.searchTerm)
    }
  }
}

export default SearchBar
