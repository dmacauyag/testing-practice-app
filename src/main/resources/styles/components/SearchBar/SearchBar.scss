.search-bar__input {
  -webkit-tap-highlight-color: transparent;
  border-radius: 10px;
  font-family: inherit;
  font-size: inherit;
  height: 100%;
  outline: none;
  text-align: center;
  width: 75%;
}
