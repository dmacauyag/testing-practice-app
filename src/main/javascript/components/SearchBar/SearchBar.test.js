import SearchBar from './SearchBar'
import React from 'react'
import renderer from 'react-test-renderer'

describe('the SearchBar component', () => {
  it('should correctly render the SearchBar component', () => {
    const component = renderer.create(
      <SearchBar />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
