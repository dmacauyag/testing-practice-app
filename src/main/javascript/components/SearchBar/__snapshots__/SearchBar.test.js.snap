// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the SearchBar component should correctly render the SearchBar component 1`] = `
<div
  className="search-bar"
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    value=""
  />
</div>
`;
